Null Avatar URL Exception On PeerTube / Youtube Subscriptions with a Blank Avatar

### Checklist

- [X] I am able to reproduce the bug with the latest version given here: [CLICK THIS LINK](https://github.com/TeamNewPipe/NewPipe/releases/latest).
- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Affected version

0.26.1

### Steps to reproduce the bug

1. Subscribe to a PeerTube / Youtube channel that does not have a user-supplied avatar (blank avatar) 
2. Scroll up and release on the "What's New" page to initate feed refresh. 

### Expected behavior

The feed finishes refreshing without throwing an exception for subscriptions with blank / default avatars.

### Actual behavior

The feed finishes loading,  and then throws a Null Pointer exception from attempting to gather information about a subscription with a blank avatar. 

### Screenshots/Screen recordings

_No response_

### Logs

## Exception
* __User Action:__ requested feed
* __Request:__ 
* __Content Country:__ US
* __Content Language:__ en-US
* __App Language:__ en_US
* __Service:__ none
* __Version:__ 0.26.1
* __OS:__ Linux samsung/a21sq/a21:12/SP1A.210812.016/A215USQUBCWG3:user/release-keys 12 - 31
<details><summary><b>Exceptions (5)</b></summary><p>
<details><summary><b>Crash log 1</b></summary><p>

```
org.schabi.newpipe.local.feed.service.FeedLoadService$RequestException: 3:https://pony.tube/video-channels/clipper
    at org.schabi.newpipe.local.feed.service.FeedLoadManager.loadStreams(FeedLoadManager.kt:229)
    at org.schabi.newpipe.local.feed.service.FeedLoadManager.access$loadStreams(FeedLoadManager.kt:33)
    at org.schabi.newpipe.local.feed.service.FeedLoadManager$startLoading$7.apply(FeedLoadManager.kt:113)
    at org.schabi.newpipe.local.feed.service.FeedLoadManager$startLoading$7.apply(FeedLoadManager.kt:112)
    at io.reactivex.rxjava3.internal.operators.parallel.ParallelMap$ParallelMapSubscriber.onNext(ParallelMap.java:116)
    at io.reactivex.rxjava3.internal.operators.parallel.ParallelFilter$ParallelFilterSubscriber.tryOnNext(ParallelFilter.java:132)
    at io.reactivex.rxjava3.internal.operators.parallel.ParallelRunOn$RunOnConditionalSubscriber.run(ParallelRunOn.java:399)
    at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:65)
    at io.reactivex.rxjava3.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:56)
    at java.util.concurrent.FutureTask.run(FutureTask.java:264)
    at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:307)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
    at java.lang.Thread.run(Thread.java:1012)
Caused by: java.lang.NullPointerException: (info as? ChannelInfo)?.…?: subscription.avatarUrl must not be null
    at org.schabi.newpipe.local.feed.service.FeedUpdateInfo.<init>(FeedUpdateInfo.kt:39)
    at org.schabi.newpipe.local.feed.service.FeedLoadManager.loadStreams(FeedLoadManager.kt:220)
    ... 13 more

```
</details>

### Affected Android/Custom ROM version

Android 12 

### Affected device model

Samsung Galaxy A21 SM-A215U

### Additional information

I have a suspicion that this issue is identical to #10778. The stack trace both resolve to the same Null Pointer Exception: `Caused by: java.lang.NullPointerException: (info as? ChannelInfo)?.…?: subscription.avatarUrl must not be null` issue. 

The fact that #10778 is caused by a failure to fetch a blank avatar from a YouTube subscription, and in this case the same exception is thrown on a PeerTube channel might indicate that there need to be more robust extractor methods to handle this scenario in the future especially if  it turns out to be a cross service issue. 

thank you so much for your time and beautiful work!!

--- a/app/src/main/java/org/schabi/newpipe/database/subscription/SubscriptionEntity.java
+++ b/app/src/main/java/org/schabi/newpipe/database/subscription/SubscriptionEntity.java
@@ -1,6 +1,7 @@
 package org.schabi.newpipe.database.subscription;
 
 import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
 import androidx.room.ColumnInfo;
 import androidx.room.Entity;
 import androidx.room.Ignore;
@@ -95,11 +96,12 @@ public class SubscriptionEntity {
         this.name = name;
     }
 
+    @Nullable
     public String getAvatarUrl() {
         return avatarUrl;
     }
 
-    public void setAvatarUrl(final String avatarUrl) {
+    public void setAvatarUrl(@Nullable final String avatarUrl) {
         this.avatarUrl = avatarUrl;
     }
 

--- a/app/src/main/java/org/schabi/newpipe/local/feed/service/FeedUpdateInfo.kt
+++ b/app/src/main/java/org/schabi/newpipe/local/feed/service/FeedUpdateInfo.kt
@@ -18,7 +18,7 @@ data class FeedUpdateInfo(
     @NotificationMode
     val notificationMode: Int,
     val name: String,
-    val avatarUrl: String,
+    val avatarUrl: String?,
     val url: String,
     val serviceId: Int,
     // description and subscriberCount are null if the constructor info is from the fast feed method

--- a/app/src/main/java/org/schabi/newpipe/local/subscription/SubscriptionManager.kt
+++ b/app/src/main/java/org/schabi/newpipe/local/subscription/SubscriptionManager.kt
@@ -100,7 +100,9 @@ class SubscriptionManager(context: Context) {
         val subscriptionEntity = subscriptionTable.getSubscription(info.uid)
 
         subscriptionEntity.name = info.name
-        subscriptionEntity.avatarUrl = info.avatarUrl
+
+        // some services do not provide an avatar URL
+        info.avatarUrl?.let { subscriptionEntity.avatarUrl = it }
 
         // these two fields are null if the feed info was fetched using the fast feed method
         info.description?.let { subscriptionEntity.description = it }

